docs/ref/databases.txt

```diff
@@ -870,6 +870,26 @@ If you're getting this error, you can solve it by:
   This will make SQLite wait a bit longer before throwing "database is locked"
   errors; it won't really do anything to solve them.
 
+.. _sqlite-transaction-mode:
+
+Setting ``transaction_mode``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You might see the "Database is Locked" errors before
+the specified timeout when working with transactions. To resolve this, change 
+the transaction mode to ``IMMEDIATE``::
```

Comment:
This doesn't explain anything to me as a user, check out other [docs](https://docs.djangoproject.com/en/5.0/ref/databases/#mysql-isolation-level) on inspiration what to say.

Reply:
I've expanded the whole doc with a lot more information. I'm struggling because the problem that IMMEDIATE solves only happens under a very particular set of conditions, but it's very common when you have writes from multiple connecitons. 

I think ideally, Django users shouldn't have to care/read about this and Django should just do the right thing. With SQLite this means enabling WAL and using IMMEDIATE. Maybe once this PR and https://github.com/django/django/pull/14824 are merged we could update the startproject template with these settings so that new projects have a good set of defaults when starting out.

Reply:
Whether it's the _right thing_ to do by default is a whole can of worms that we cannot assume here.

We haven't even run the whole test suite with these options enabled to confirm it doesn't break anything, setup CI to continiously do so, and confirm there is an upgrade path for the 10+ years old projects out there using SQLite if any problem is discovered.

It's one thing to provide an option, which this PR and the associated docs should focus on, and a completely different one to make it the default, which we can't assume here.

Reply:
Yes, sorry if I was unclear. I'm not proposing to change the default. I don't want to change how 10+ year old projects are configured.

I only wanted to mention that in the future, we could discuss enabling IMMEDIATE + WAL for **new** projects (the settings.py file generated for new projects would have these settings) - so that they have an SQLite config that's optimized for multiple processes/threads and not an embedded system.

Sorry again for derailing the conversation.